Documentation/ceph-object-bucket-claim-crd.md

```diff
@@ -0,0 +1,62 @@
+---
+title: Object Bucket Claim CRD
+weight: 2800
+indent: true
+---
+
+# Ceph Object Bucket Claim CRD
+
+Rook allows creation and customization of object bucket claim through the custom resource definitions (CRDs). The following settings are available for Ceph Object Bucket Claim. It also requires storage class
+
+## Sample
+##### Stoarge class
+```yaml
+apiVersion: storage.k8s.io/v1
+kind: StorageClass
+metadata:
+   name: rook-ceph-bucket
+provisioner: ceph.rook.io/bucket
+# set the reclaim policy to delete the bucket and all objects
+# when its OBC is deleted.
+reclaimPolicy: Delete
+parameters:
+  objectStoreName: my-store
+  objectStoreNamespace: rook-ceph
+  region: us-east-1
+  #bucketName:
+```
+##### Object Bucket Claim 
+```yamlapiVersion: objectbucket.io/v1alpha1
+kind: ObjectBucketClaim
+metadata:
+  name: ceph-bucket
+spec:
+  #bucketName: 
+  generateBucketName: ceph-bkt
+  storageClassName: rook-ceph-bucket
+```
+
+## Object Bucket Claim Settings
+
+### Storage Class
+
+##### Metadata
+
+- `name`: The name of the storage class need to create, which will be reflected in the object bucket claim.
+- `reclaimpolicy`: two policies supported `Delete` and `Retain`, see explaination as [specified in Kubernetes](https://kubernetes.io/docs/concepts/storage/persistent-volumes/#retain)
+
+##### Parameters
+- `objectStoreName` : Name of the object store
+- `objectStoreNamespace` : Name space where object store resides
+- `region`: Region where object store resides
+- `bucketName`: If Object Bucket Claim need to use already existing buckets
+
+### Object Bucket Claim
+
+##### Metadata
+- `name`: name of object bucket claim, with the same the required object bucket, secret and config-map will be created 
```

Comment:
A bit confusing.  Could be changed to

`OBC name - reused for Object Bucket, Secret, and ConfigMap naming`

Reply:
Done